src/python/grpcio/grpc/__init__.py

```diff
@@ -1252,6 +1269,65 @@ def ssl_server_credentials(private_key_certificate_chain_pairs,
             ], require_client_auth))
 
 
+def ssl_server_credentials_with_cert_config_fetcher(cert_config_fetcher_cb,
+                                                    require_client_auth=False):
+    """Creates a ServerCredentials for use with an SSL-enabled Server.
+
+    Args:
+      cert_config_fetcher_cb (callable): a callback that takes no
```

Comment:
Why one parameter that must behave in two different ways depending on context rather than two different parameters each subject to simpler constraints?

Reply:
are you referring to the fact that during server initialization the callback must return a valid config, vs subsequently it can return a valid config or `None`?
if so, that's because it is required by the C-core: during sever initialization, the C-core has to call the callback to obtain the certificate config, and C-core will error without a valid cert config returned. https://github.com/grpc/grpc/blob/master/include/grpc/grpc_security.h#L396

Reply:
in the two different ways i've implemented the callback (in the `interop` test and in a separate example for my own use case), it doesn't seem that complex. essentially the callback has to determine: "do i have new cert info that i want the server to use, if so, return the new cert info, else return None", and the initialization case is a trivial check.

nevertheless, do you mind specifying the interface you have in mind? is it: python grpc asks for two callbacks: one will be called only once for initialization, and the other will be called for every new client connection? the user can of course choose to, for example, provide the same callable object for both.

thanks!

Reply:
Yes, it is "[gRPC Python] asks for two callbacks: one will be called only once for initialization, and the other will be called for every new client connection".

Reply:
Do you want the two callbacks API no matter what, i.e., regardless of the discussion in https://github.com/grpc/grpc/pull/13188#issuecomment-342924015, or it depends on the outcome of that discussion?
If the former (i.e., no matter what), then I'll start on that right away instead of waiting for the discussion :)
Thanks!

Reply:
Do you think it would help to meet on google hangout for a quick 15 minutes or so to iron out the issue? If you can do that, please let me know a few time slots that works for you and I'll schedule the hangout. Thanks!

Reply:
hmm i guess https://github.com/grpc/grpc/pull/13188#pullrequestreview-75202684 says you `requested changes` i guess you want the API with two callbacks regardless of the outcome of this discussion. so i'll go ahead and make the change.

Reply:
Yes, let's absolutely hang out. How's your tomorrow (Friday)? I'm available 8-11:30 in the morning and 12:30-3:30 in the afternoon.

Reply:
One question I'd like us to address: why pass an `initial_cert_config_fetcher` rather than pass an `initial_cert_config`? Why do this dance of indirection rather than just pass the data?